### PR TITLE
Ability to update time entry

### PR DESCRIPTION
**Pull request by sprzenus**

### CODE REVIEW

File: TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift [hunk 1]
```diff
@@ -156,6 +154,24 @@ class WorkTimesViewModel: WorkTimesViewModelType {
...
     
+    private func createTask(for indexPath: IndexPath) -> Task? {
+        guard let dailyWorkTime = dailyWorkTime(for: indexPath) else { return nil }
+        guard let workTime = workTime(for: indexPath) else { return nil }
+        let url: URL?
+        if let taskUrlString = workTime.task {
+            url = URL(string: taskUrlString)
+        } else {
+            url = nil
+        }
```
Comment (PiotrPawlus) [inline TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift:165]: You don't have to set `URL` to nil. It is already nil

File: TimeTable/Categories/WorkTime/WorkTimeViewModel.swift [hunk 2]
```diff
@@ -182,8 +188,20 @@ class WorkTimeViewModel: WorkTimeViewModelType {
         guard fromDate < toDate else { throw UIError.timeGreaterThan }
     }
     
+    private func handleUpdateSuccess() {
+        self.userInterface?.dismissView()
+    }
+    
+    private func handleUpdateFailure(withError error: Error) {
+        self.errorHandler.throwing(error: error)
```
Comment (PiotrPawlus) [inline TimeTable/Categories/WorkTime/WorkTimeViewModel.swift]: It is only used on update failure?

File: TimeTable/Categories/WorkTime/WorkTimeViewModel.swift [hunk 3]
```diff
@@ -182,8 +188,20 @@ class WorkTimeViewModel: WorkTimeViewModelType {
         guard fromDate < toDate else { throw UIError.timeGreaterThan }
     }
     
+    private func handleUpdateSuccess() {
+        self.userInterface?.dismissView()
```
Comment (PiotrPawlus) [inline TimeTable/Categories/WorkTime/WorkTimeViewModel.swift]: It is only used on update success?

File: TimeTable/Models/ApiClient/ApiClient.swift [hunk 4]
```diff
@@ -101,4 +103,31 @@ class ApiClient: ApiClientNetworkingType {
     func delete(_ endpoint: Endpoints, completion: @escaping ((Result<Void>) -> Void)) {
         networking.delete(endpoint.value, completion: completion)
     }
+    
+    func put<E: Encodable, D: Decodable>(_ endpoint: Endpoints, parameters: E?, completion: @escaping ((Result<D>) -> Void)) {
+        do {
```
Comment (PiotrPawlus) [inline TimeTable/Models/ApiClient/ApiClient.swift:108]: You're missing unit tests for 
```
func put<E: Encodable, D: Decodable>(_ endpoint: Endpoints, parameters: E?, completion: @escaping ((Result<D>) -> Void))
```

File: TimeTable/Models/ApiClient/ApiClient.swift [hunk 5]
```diff
@@ -101,4 +103,31 @@ class ApiClient: ApiClientNetworkingType {
...
+            _ = networking.put(endpoint.value, parameters: parameters) { [weak self] response in
+                self?.handle(response: response, completion: completion)
+            }
+        } catch {
+            completion(.failure(ApiClientError(type: .invalidParameters)))
+        }
+    }
+    
+    func put<E: Encodable>(_ endpoint: Endpoints, parameters: E?, completion: @escaping ((Result<Void>) -> Void)) {
+        do {
```
Comment (PiotrPawlus) [inline TimeTable/Models/ApiClient/ApiClient.swift:119]: You're missing unit tests for 
```
func put<E: Encodable>(_ endpoint: Endpoints, parameters: E?, completion: @escaping ((Result<Void>) -> Void))
```

File: TimeTable/Models/BussinessModels/WorkTimes/ApiClient+WorkTimes.swift [hunk 6]
```diff
@@ -27,4 +28,8 @@ extension ApiClient: ApiClientWorkTimesType {
     func deleteWorkTime(identifier: Int64, completion: @escaping ((Result<Void>) -> Void)) {
         delete(.workTime(identifier), completion: completion)
     }
+    
+    func updateWorkTime(identifier: Int64, parameters: Task, completion: @escaping ((Result<Void>) -> Void)) {
+        put(.workTime(identifier), parameters: parameters, completion: completion)
```
Comment (PiotrPawlus) [inline TimeTable/Models/BussinessModels/WorkTimes/ApiClient+WorkTimes.swift:33]: You're missing unit tests for 
```
 func updateWorkTime(identifier: Int64, parameters: Task, completion: @escaping ((Result<Void>) -> Void))
```

File: TimeTable/Coordinators/TimeTable/WorkTime/WorkTimesCoordinator.swift [hunk 7]
```diff
@@ -56,24 +57,40 @@ class WorkTimesCoordinator: BaseNavigationCoordinator, BaseTabBarCordninatorType
...
+                                          errorHandler: errorHandler,
+                                          calendar: Calendar.autoupdatingCurrent,
+                                          lastTask: lastTask,
+                                          editedTask: nil)
+        controller.configure(viewModel: viewModel, notificationCenter: NotificationCenter.default)
+        showWorkTimeController(controller: controller, sourceView: sourceView)
+    }
+    
+    func workTimesRequestedForEditWorkTimeView(sourceView: UIView, editedTask: Task) {
+        guard let controller: WorkTimeViewControlleralbe = storyboardsManager.controller(storyboard: .workTime, controllerIdentifier: .initial) else { return }
```
Comment (PiotrPawlus) [inline TimeTable/Coordinators/TimeTable/WorkTime/WorkTimesCoordinator.swift:86]: You're missing unit tests for 
```
func workTimesRequestedForEditWorkTimeView(sourceView: UIView, editedTask: Task)
```

File: TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift [hunk 8]
```diff
@@ -134,18 +135,15 @@ class WorkTimesViewModel: WorkTimesViewModelType {
...
-                   url: nil,
-                   day: dailyWorkTimesArray.first?.day,
-                   startAt: lastWorkTime?.startsAt,
-                   endAt: lastWorkTime?.endsAt)
+        let lastTask = createTask(for: IndexPath(row: 0, section: 0))
         coordinator.workTimesRequestedForNewWorkTimeView(sourceView: sourceView, lastTask: lastTask)
     }
     
+    func viewRequestedForEditEntry(sourceView: UITableViewCell, at indexPath: IndexPath) {
+        guard let task = createTask(for: indexPath) else { return }
```
Comment (PiotrPawlus) [inline TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift:143]: You're missing unit test for:
```
func viewRequestedForEditEntry(sourceView: UITableViewCell, at indexPath: IndexPath)
```

File: TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift [hunk 9]
```diff
@@ -156,6 +154,24 @@ class WorkTimesViewModel: WorkTimesViewModelType {
         return dailyWorkTimesArray[indexPath.section]
     }
     
+    private func createTask(for indexPath: IndexPath) -> Task? {
+        guard let dailyWorkTime = dailyWorkTime(for: indexPath) else { return nil }
```
Comment (PiotrPawlus) [inline TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift:158]: Please add unit tests for this method

File: TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift (hunk 1 above)
Comment (sprzenus) [reply TimeTable/Categories/WorkTimes/WorkTimesViewModel.swift:165]: It is not initialized then. url is a constant

**Pull request merged**